fuse_loss/include/fuse_loss/arctan_loss.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Locus Robotics
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef FUSE_LOSS_ARCTAN_LOSS_H
+#define FUSE_LOSS_ARCTAN_LOSS_H
+
+#include <fuse_core/loss.h>
+
+
+namespace fuse_loss
+{
+
+/**
+ * @brief The ArctanLoss loss function.
+ *
+ * This class encapsulates the ceres::ArctanLoss class, adding the ability to serialize it and load it dynamically.
+ *
+ * See the Ceres documentation for more details. http://ceres-solver.org/nnls_modeling.html#lossfunction
+ */
+class ArctanLoss : public fuse_core::Loss
```

Comment:
This and the other `Loss` children classes are very thin wrappers on top of the corresponding `ceres::LossFunction`. I refrained from creating a pre-processor macro to instantiate them because I think that would have been ugly, especially because the `TolerantLoss` function has two arguments instead of one, which is what most loss functions have. Do you think there's a better approach? TBH I don't expect the loss functions to change or the number of them grow, so I think this should be fine. However, it's true there's a lot of code duplication.

Reply:
This is how I would have implemented it as well. All of the code here is just boilerplate stuff; no "real work" is duplicated. And the fact that the various loss functions have only one parameter, and that parameter is called `a` is more coincidence than by design. The different loss functions could have easily had different parameter names; then the code duplication would look a lot more necessary.

Reply:
:+1: